typed_sql/lib/src/typed_sql.annotations.dart

```diff
@@ -0,0 +1,80 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+part of 'typed_sql.dart';
+
+/// Annotation for a table specifying it's primary key.
+final class PrimaryKey {
+  final List<String> fields;
+
+  const PrimaryKey(this.fields);
+}
+
+/// Annotation for field that has a default value, specified by [value].
+final class DefaultValue<T> {
+  final T value;
+
+  const DefaultValue(this.value);
+}
+
+/// Annotation for field that should be auto-incremented by default.
```

Comment:
Isn't "by default" implied by "auto"?
```suggestion
/// Annotation for a field that should be auto-incremented.
```